metadata.json

```diff
@@ -11,7 +11,7 @@
   "requirements": [
     {
       "name": "puppet",
-      "version_requirement": ">= 4.7.1 < 6.0.0"
+      "version_requirement": ">= 4.7.1 < 7.0.0"
```

Comment:
do the new functions actually work with puppet 4.7.1 or do they require puppet5/4.10?

Reply:
In the docs it states that the legacy API should only be used for puppet 3.x compatibility, so I assumed that puppet 4.x should be supported. The [docs for 4.9](https://puppet.com/docs/puppet/4.9/functions_ruby_overview.html) mention both APIs so at least 4.9 should be supported (these pages do not exist for 4.8 and 4.7 unfortunately).